### PR TITLE
Add scoring logic inspired by npms.io work

### PR DESCRIPTION
**Pull request by kevmoo**
CC @isoos

### CODE REVIEW

File: lib/src/scoring.dart [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    var total = 0.0;
+    num max, min;
+
+    var count = 0;
+    for (var i in numbers) {
+      //TODO(kevmoo): validate i for Nan/Infinity
+      count++;
+      total += i;
+
+      if (max == null || i > max) {
```
Comment (isoos) [inline lib/src/scoring.dart]: import 'dart:math' show min, max;
num maxValue = numbers.fold(numbers.first, max);
num minValue = numbers.fold(numbers.first, min);
(or use the prefixed lib if you keep the min/max field names)

File: lib/src/scoring.dart [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    }
+  }
+
+  // Convert back from depressed cubic
+  for (var i = 0; i < roots.length; i++) roots[i] -= b / (3 * a);
+
+  return roots;
+}
+
+num abs(a) => a < 0 ? -a : a;
```
Comment (isoos) [inline lib/src/scoring.dart]: why not a.abs() ?
https://api.dartlang.org/stable/1.23.0/dart-core/num/abs.html

File: lib/src/scoring.dart (hunk 1 above)
Comment (kevmoo) [inline lib/src/scoring.dart]: then we iterate twice – and then a third time to count the items. I designed it to allow a streamed input.

File: lib/src/scoring.dart (hunk 2 above)
Comment (kevmoo) [inline lib/src/scoring.dart]: Oh...yeah... 😊

File: lib/src/scoring.dart (hunk 1 above)
Comment (isoos) [inline lib/src/scoring.dart]: right...
the if can be spared though with:
max = math.max(i, max ?? i);

but not that important

Comment (kevmoo) [inline lib/src/scoring.dart]: nice!

**Comment by kevmoo** [comment]
@isoos PTAL

**Comment by isoos** [comment]
This got mixed commits with the "Use dart files to store the expected output". I'm actually fine with that too, so merging...

**Comment by kevmoo** [comment]
In the future, let me do the merge after you approve. I like clean history.
:-)

**Pull request merged**